src/renderer/vt/paint.cpp

```diff
@@ -535,8 +535,17 @@ using namespace Microsoft::Console::Types;
     // Move the cursor to the start of this run.
     RETURN_IF_FAILED(_MoveCursor(coord));
 
-    // Write the actual text string
-    RETURN_IF_FAILED(VtEngine::_WriteTerminalUtf8({ _bufferLine.data(), cchActual }));
+    // Write the actual text string. If we're using a soft font, the character
+    // set should have already been selected, so we just need to map our internal
+    // representation back to ASCII (handled by the _WriteTerminalDrcs method).
+    if (_usingSoftFont)

```

Comment:
```suggestion
    if (_usingSoftFont) [[unlikely]]
```

@lhecker is this the correct usage of that annotation in C++?

Reply:
Yeah that would work. likely/unlikely aren't particularly impactful in MSVC right now, but it does influence the inalienability of functions quite a bit, which can help make our binaries more compact.

Reply:
I've just tried this, and it does make a bit of a difference. Both of the `_WriteTerminalXXX` calls are inline, so there is quite a lot of code in the two branches of that condition. With the `unlikely` attribute added, part of the first branch gets pulled out of the main body of code, so there is a slightly shorter jump to the else branch.

I don't know if that's really worth the effort, though. I think if you wanted to make a significant difference with these attributes, a good place to put them might be in the `RETURN_IF_FAILED` macros. That should remove all that diagnostic logging from the main path of execution which is probably quite a chunk of code.

I'm not an optimization expert though, so I'll leave it to you guys to decide.

Reply:
Only saw Leonard's reply after I posted mine. I've applied the suggested change now.